hibernate-core/src/main/java/org/hibernate/boot/internal/SessionFactoryOptionsBuilder.java

```diff
@@ -975,7 +975,7 @@ public boolean isFailOnPaginationOverCollectionFetchEnabled() {
 
 	@Override
 	public JpaCompliance getJpaCompliance() {
-		return jpaCompliance;
+		return jpaCompliance.immutableCopy();
```

Comment:
Is this only ever called once?  If not, maybe we want to change this so we do only call it once.  Doubt it has a huge perf impact, but every bit helps